paasta_tools/kubernetes_tools.py

```diff
@@ -2537,30 +2524,41 @@ def update_stateful_set(
     )
 
 
-@a_sync.to_async
-def get_events_for_object(
+def get_event_timestamp(event: V1Event) -> Optional[float]:
+    # Cycle through timestamp attributes in order of preference
+    for ts_attr in ["last_timestamp", "event_time", "first_timestamp"]:
+        ts = getattr(event, ts_attr)
+        if ts:
+            return ts.timestamp()
+    return None
+
+
+@async_timeout()
+async def get_events_for_object(
     kube_client: KubeClient,
     obj: Union[V1Pod, V1Deployment, V1StatefulSet, V1ReplicaSet],
     kind: str,  # for some reason, obj.kind isn't populated when this function is called so we pass it in by hand
+    max_age: int = 0,
```

Comment:
Originally I set this as 0 rather than None since it simplifies the conditional for handling an invalid value (negative seconds) or no value (i.e. `if max_age is not None and max_age > 0` vs just `if max_age > 0`.  

I can switch to None if that's the better practice so we're consistent.